centreon/src/Utility/EnvironmentFileManager.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Utility;
+
+class EnvironmentFileManager
+{
+    /** @var array<string, int|float|bool|string> */
+    private array $variables = [];
+
+    private ?string $currentEnvironnementFile = null;
+
+    /**
+     * @param string $environmentFilePath
+     */
+    public function __construct(private string $environmentFilePath)
+    {
+        $this->environmentFilePath = $this->addDirectorySeparatorIfNeeded($this->environmentFilePath);
+    }
+
+    /**
+     * @param string $environmentFile
+     *
+     * @throws \Exception
+     */
+    public function load(string $environmentFile = '.env'): void
+    {
+        $this->currentEnvironnementFile = $environmentFile;
+        $filePath = $this->environmentFilePath . $environmentFile;
+        if (! file_exists($filePath)) {
+            throw new \Exception(sprintf("The environment file '%s' does not exist", $environmentFile));
+        }
+
+        $file = fopen($filePath, 'r');
+        if (! $file) {
+            throw new \Exception(sprintf('Impossible to open file \'%s\'', $filePath));
+        }
+        try {
+            while (($line = fgets($file, 1024)) !== false) {
+                $line = trim($line);
+                if (str_contains($line, '=')) {
+                    [$key, $value] = explode('=', $line);
+                    $this->add($key, $value);
```

Comment:
```suggestion
                    $this->add(trim($key), trim($value));
```
Or insite the add function

Its to avoid the keys/values with witespace